docs/dns-minikube.md

```diff
@@ -0,0 +1,170 @@
+$ cat configmap-2.yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: metallb-config
+data:
+  config: |
+    peers:
+    - peer-address: 10.0.0.1
+      peer-asn: 64501
+      my-asn: 64500
+    address-pools:
+    - name: default
+      protocol: bgp
+      addresses:
+      - 172.168.20.0/24
+
+$ kubectl apply --context cluster1 -f configmap-1.yaml
+$ kubectl apply --context cluster2 -f configmap-2.yaml
+```
+### Creating service resources
+After metallb works, create sample deployment and service from [sample](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md). Make service as LoadBalancer type.
+```
+sed -i 's/NodePort/LoadBalancer/' example/sample1/federatedservice-template.yaml
+```
+
+Create ServiceDNSRecord to make DNS work for service.
+```
+$ cat <<EOF | kubectl create -f -
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: Domain
+metadata:
+  # Corresponds to <federation> in the resource records.
+  name: test-domain
+  # The namespace running federation-controller-manager.
+  namespace: federation-system
+# The domain/subdomain that is setup in your externl-dns provider.
+domain: example.org
+---
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: ServiceDNSRecord
+metadata:
+  # The name of the sample service.
+  name: test-service
+  # The namespace of the sample deployment/service.
+  namespace: test-namespace
+spec:
+  # The name of the corresponding `Domain`.
+  domainRef: test-domain
+  recordTTL: 300
+EOF
+```
+### Enable the ingress controller
+```
+$ minikube -p cluster1 addons enable ingress
+$ minikube -p cluster2 addons enable ingress
+```
+
+After ingress controller enabled, create sample deployment, service and ingress from [sample](https://github.com/kubernetes-sigs/federation-v2/blob/master/docs/ingressdns-with-externaldns.md). Make ingress with "example.org" for this tutorial example.
+```
+sed -i 's/example.com/example.org/' example/sample1/federatedingress-template.yaml
+```
+
+### Creating ingress resources
+Create IngressDNSRecord to make DNS work for ingress.
+```
+$ cat <<EOF | kubectl create -f -
+apiVersion: multiclusterdns.federation.k8s.io/v1alpha1
+kind: IngressDNSRecord
+metadata:
+  name: test-ingress
+  namespace: test-namespace
+spec:
+  hosts:
+  - ingress.example.org
+  recordTTL: 300
+EOF
+```
+
+## DNS Example
+Wait a moment until DNS has the ingress/service IP. The DNS service IP is from CoreDNS service. It is "my-coredns-coredns" in this example.
+```
+$ kubectl get svc my-coredns-coredns
+NAME                 TYPE        CLUSTER-IP     EXTERNAL-IP   PORT(S)   AGE
+my-coredns-coredns   ClusterIP   10.100.4.143   <none>        53/UDP    12m
+
+$ kubectl run -it --rm --restart=Never --image=infoblox/dnstools:latest dnstools
+dnstools# dig @10.100.4.143 test-service.test-namespace.test-domain.svc.example.org +short
+192.168.20.0
+172.168.20.0
+dnstools# dig @10.100.4.143 ingress.example.org +short
+10.0.2.15
```

Comment:
Why only one IP here?

Reply:
There is one IP because both clusters ingress controller assigns same IP for the case.
I can try to make it with different IP in each cluster. However, it seems not mandatory for testing. Any comments?

Reply:
trace this issue in below comments for same place.